src/Common/src/Interop/Windows/Winsock/Interop.TransmitFile.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Mswsock
+    {
+        [DllImport(Interop.Libraries.Mswsock, SetLastError = true)]
+        internal static extern bool TransmitFile(
+            SafeHandle socket,
+            SafeHandle fileHandle,
+            int numberOfBytesToWrite,
+            int numberOfBytesPerSend,
+            SafeHandle overlapped,
+            TransmitFileBuffers buffers,
+            TransmitFileOptions flags);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal class TransmitFileBuffers
```

Comment:
Actually I'm not sure on the right way to do this.  What's the best approach for a structure like this than can be null?